### ISSUE TITLE
False positive for Style/TrailingComma cop

### ISSUE THREAD

**Issue by sferik** [bbatsov/rubocop#1955]
In RuboCop version 0.32.0, given the following configuration:

```
Style/TrailingComma:
  EnforcedStyleForMultiline: 'comma'
```

And the following code:

```ruby
Foo.new({})
```

RuboCop generates the following offense:

```
Put a comma after the last parameter of a multiline method call.
Foo.new({})
        ^^
```

With autocorrect enabled, the code is changed to:

```ruby
Foo.new({},)
```

Something is clearly wrong. This was not a problem in earlier versions of RuboCop.

**Comment by mattjmcnaughton** [comment]
I made a pull request with a possible fix. Basically the cop interpreted the expression, correctly, as a method call. As can be seen [in the code](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/trailing_comma.rb#L78), because the last argument, the `{}` has no comma, and `comma` is the setting, an offense was reported. 

I simply added a check to not report an offense if the first argument of the function is an empty hash, as can be seen [here](https://github.com/bbatsov/rubocop/pull/1956/files#diff-f1ee3cd2583ac993bde9cd52883ba3beR80).

My only concern, as mentioned on the [pull request](https://github.com/bbatsov/rubocop/pull/1956#discussion-diff-31888224R80) is that my solution would create false negatives. For example, it would not report an offense for `Foo.new({}, ... \n /// 5)`, when perhaps it should.

Thoughts?

**Comment by sferik** [comment]
The error message says:

> Put a comma after the last parameter of a multiline method call.

IMHO, it should only apply to **multiline** method calls. For example:

```ruby
Foo.new(
  "first arg",
  "second arg",
)
```

**Comment by mattjmcnaughton** [comment]
Hmm so maybe the problem is that `Foo.new({})` should never have been called multiline in the first place. I'm guessing that has something to do with the new definition of multiline (seen [here](https://github.com/bbatsov/rubocop/blob/121fdb2fea1bcf40083a94711c3195e5434d1b74/lib/rubocop/cop/style/trailing_comma.rb#L122)), although I may be totally wrong about that.

**Comment by philoserf** [comment]
I thought we argued this style through and decided against this default. That we are encountering this again here is a bloody nose to rubocop. 

IMO, rubocop should not default to disputed standards. If the community standard is unsettled no rubocop default setting should enforce a standard even if it is override able in settings. Out of the box behavior should be non controversial. 

Or a close to possible as a community standard can be. This style had real opposition.

**Comment by sferik** [comment]
@philoserf This is not the default. I am overwriting the default, as I say in the first line of the description with the configuration:

```
Style/TrailingComma:
  EnforcedStyleForMultiline: 'comma'
```

**Comment by philoserf** [comment]
My miss. Comment retracted.

**Comment by mattjmcnaughton** [comment]
I updated my [pull request](https://github.com/bbatsov/rubocop/pull/1956) off of @sferik's observation that this method should not have been labelled `multiline?` in the first place. Let me know if I can make any changes to improve it!

**Comment by jfelchner** [comment]
My setup is exactly like @sferik and I'm also getting this problem.

**Comment by ddfreyne** [comment]
I’m running into the same exact issue.

Here’s another case where the cop gives incorrect output:

```ruby
Foo.new(
  'blah blah',
  { owner: 'me', author: 'myself' },
  'more blah',
)
```

RuboCop reports:

```
stuff.rb:6:14: C: Avoid comma after the last parameter of a method call, unless each item is on its own line.
  'more blah',
             ^
```
